src/Asset/Capacity/HasSoftwaresCapacity.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Asset\Capacity;
+
+use CommonGLPI;
+use Item_SoftwareLicense;
+use Item_SoftwareVersion;
+use Session;
+use Software;
+use SoftwareLicense;
+use SoftwareVersion;
+
+class HasSoftwaresCapacity extends AbstractCapacity
+{
+    public function getLabel(): string
+    {
+        return Software::getTypeName(Session::getPluralNumber());
+    }
+
+    public function getCloneRelations(): array
+    {
+        return [
+            Item_SoftwareVersion::class,
+            Item_SoftwareLicense::class,
+        ];
+    }
+
+    public function isUsed(string $classname): bool
+    {
+        return parent::isUsed($classname)
+            && $this->countAssetsLinkedToPeerItem($classname, Item_SoftwareVersion::class) > 0;
```

Comment:
People can link asset to software licenses without linking them to software versions (the opposite is also true).

```suggestion
            && (
                $this->countAssetsLinkedToPeerItem($classname, Item_SoftwareVersion::class) > 0
                || $this->countAssetsLinkedToPeerItem($classname, Item_SoftwareLicense::class) > 0
            );
```

This should also be tested. If it is not possible to do it using the existing `provideIsUsed/testIsUsed` methods, then just declare an `provideIsUsed()` method in test class and do the test in a method with another name (e.g. `testIsUsedCustom`).